src/wolfsftp.c

```diff
@@ -2817,43 +2794,25 @@ static int wolfSSH_SFTPNAME_readdir(WOLFSSH* ssh, WDIR* dir, WS_SFTPNAME* out,
 
     /* attempt to get file attributes. Could be directory or have none */
     {
-        char* buf;
-        int   bufSz;
-        int   tmpSz;
+        char  r[WOLFSSH_MAX_FILENAME];
+        char  s[WOLFSSH_MAX_FILENAME];
 
-        bufSz = out->fSz + (int)WSTRLEN(dirName) + sizeof(WS_DELIM);
-        buf = (char*)WMALLOC(bufSz + 1, out->heap, DYNTYPE_SFTP);
-        if (buf == NULL) {
-            WFREE(out->fName, out->heap, DYNTYPE_SFTP);
-            return WS_MEMORY_E;
-        }
-        buf[0] = '\0';
-        WSTRNCAT(buf, dirName, bufSz + 1);
-        tmpSz = (int)WSTRLEN(buf);
+        r[0] = '\0';
+        WSTRNCAT(r, dirName, sizeof(r));
```

Comment:
Same question as earlier about this pattern.

Reply:
Returning an error if everything won't fit into r, and freeing the out filename.